pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -149,206 +145,142 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 	if errors.IsNotFound(err) {
 		// The resource no longer exists, in which case we stop processing.
 		c.Logger.Errorf("pipeline run %q in work queue no longer exists", key)
-		return nil
+		return controller.NewPermanentError(err)
 	} else if err != nil {
 		return err
 	}
 
 	// Don't modify the informer's copy.
 	pr := original.DeepCopy()
+
 	if !pr.HasStarted() {
+		// HasStarted is made false by InitializeConditions
```

Comment:
i'm kind of confused by this comment - especially since we just called HasStarted and it's already false - would it make sense to elaborate a bit more or remove this comment?